lsp/core/aws-lsp-core/src/content/handlers/cachedContentHandler.ts

```diff
@@ -0,0 +1,91 @@
+import { URI } from 'vscode-uri'
+import { DefaultHttpRequester, HttpRequester, getETag } from '../../http/requesters'
+import { TimeProvider } from '../../util/timeProvider'
+import { ContentMetadata, UriCacheRepository } from '../cache/uriCacheRepository'
+import {
+    ContentRequestMiddleware,
+    ContentRequestMiddlewareDelegate,
+    ContentRequestResponse,
+} from '../contentRequestMiddleware'
+
+export interface CachedContentHandlerProps {
+    cacheRepository: UriCacheRepository
+    httpRequester?: HttpRequester
+    timeProvider?: TimeProvider
+}
+
+/**
+ * Middleware that will use a cached version of content if the online version has not been updated.
+ * Newer content that has been downloaded will be cached.
+ *
+ * This avoids unnecessary downloads.
+ *
+ * Cached content is considered stale after {@link thirtyMinutesInMillis}
+ * and is refreshed.
+ */
+export class CachedContentHandler implements ContentRequestMiddleware {
```

Comment:
Does this have logging or some other visibility into its decisions for troubleshooting

Reply:
TBD - We currently aren't using this component, and won't be for a while.